jax/_src/scipy/linalg.py

```diff
@@ -1041,3 +1041,49 @@ def toeplitz(c: ArrayLike, r: Optional[ArrayLike] = None) -> Array:
       (nrows,), (1,), 'VALID', dimension_numbers=('NTC', 'IOT', 'NTC'),
       precision=lax.Precision.HIGHEST)[0]
   return jnp.flip(patches, axis=0)
+
+
+_NULLSPACE_DOC = """\
+Because the size of the output of ``null_space`` is data-dependent, the function is not
+typically compatible with JIT. The JAX version adds the optional ``size`` argument which
+must be specified statically for ``jax.scipy.linalg.null_space`` to be used within some of JAX's
+transformations.
+"""
+_NULLSPACE_EXTRA_PARAMS = """
+size : int, optional
+    If specified, ``size`` elements of the null space are returned. If there are
+    fewer null space elements than ``size`` indicates, the return value will be padded with constant
+    ``fill_value`` vectors.
+fill_value : array_like, optional
+    When ``size`` is specified and there are fewer null space elements than the indicated number of elements,
+    the return value will be padded with constant ``fill_value`` vectors. If ``fill_value`` is not specified,
+    it defaults to zero (indeed constant 0 vectors are always in the null space).
+"""
+
+@_wraps(scipy.linalg.null_space, lax_description=_NULLSPACE_DOC, extra_params=_NULLSPACE_EXTRA_PARAMS)
+def null_space(A: ArrayLike, rcond: Optional[float] = None,
+               size: Optional[int] = None, fill_value: Union[None, ArrayLike, Tuple[ArrayLike]] = None) -> Array:
+  u, s, vh = svd(A, full_matrices=True)
+  M, N = u.shape[0], vh.shape[1]
+  if rcond is None:
+    rcond = jnp.finfo(s.dtype).eps * max(M, N)
+  tol = jnp.amax(s) * rcond
+  num = jnp.sum(s > tol, dtype=int)
+  if size is None:
+    size = int(num)
```

Comment:
This is the line that will error if `size=None` is used under `jit`. It may be helpful to use a more informative error message; something like this: https://github.com/google/jax/blob/fdc8864d9b0ade449a66933ee7fe0bb3dd27ede3/jax/_src/numpy/lax_numpy.py#L1371-L1373

Reply:
oh right will do so

Reply:
done